python/srf/_pysrf/include/pysrf/edge_adapter.hpp

```diff
@@ -0,0 +1,247 @@
+/**
+ * SPDX-FileCopyrightText: Copyright (c) 2021-2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <pysrf/types.hpp>  // IWYU pragma: keep
+#include <pysrf/utils.hpp>
+
+#include <srf/channel/forward.hpp>
+#include <srf/channel/ingress.hpp>
+#include <srf/channel/status.hpp>
+#include <srf/node/edge.hpp>
+#include <srf/node/edge_adapter_registry.hpp>
+#include <srf/node/edge_builder.hpp>
+#include <srf/node/edge_connector.hpp>
+#include <srf/node/edge_registry.hpp>
+#include <srf/node/forward.hpp>
+#include <srf/node/rx_node.hpp>
+#include <srf/node/rx_sink.hpp>
+#include <srf/node/rx_source.hpp>
+#include <srf/node/sink_properties.hpp>
+#include <srf/node/source_properties.hpp>
+
+#include <glog/logging.h>
+#include <pybind11/cast.h>
+#include <pybind11/gil.h>
+#include <pybind11/pybind11.h>  // IWYU pragma: keep
+#include <pybind11/pytypes.h>
+#include <rxcpp/rx-observable.hpp>
+#include <rxcpp/rx-subscriber.hpp>
+#include <rxcpp/rx-subscription.hpp>
+
+#include <functional>  // for function, ref
+#include <memory>      // for shared_ptr, __shared_ptr_access, dynamic_pointer_cast, allocator, make_shared
+#include <ostream>     // for operator<<
+#include <type_traits>
+#include <typeindex>  // for type_index
+#include <utility>    // for move, forward:
+
+namespace srf::pysrf {
+
+/**
+ * @brief Utility struct which supports building pySRF source/sink adapter functions which can be registered with
+ * the EdgeAdapterRegistry.
+ */
+struct EdgeAdapterUtil
```

Comment:
should the file name be `edge_adaptor_util`?

Reply:
Possibly, although I left it a bit more ambiguous because of the AutoRegXXX structures there.